bundle/edu.gemini.spdb.reports.collection/src/test/java/edu/gemini/dbTools/ephemeris/EphemerisFilesTest.scala

```diff
@@ -0,0 +1,247 @@
+package edu.gemini.dbTools.ephemeris
+
+import edu.gemini.spModel.core.AlmostEqual._
+
+import edu.gemini.spModel.core.HorizonsDesignation
+import edu.gemini.spModel.core.HorizonsDesignation.MajorBody
+
+import java.nio.file.{Files, Path}
+import java.util.logging.{Level, Logger}
+
+import org.scalacheck.Prop
+import org.specs2.ScalaCheck
+import org.specs2.mutable.Specification
+
+import scala.collection.JavaConverters._
+
+import scalaz._, Scalaz._
+
+object EphemerisFilesTest extends Specification with ScalaCheck with Arbitraries {
+  val log = Logger.getLogger(EphemerisFilesTest.getClass.getName)
+
+  def delete(f: Path): Unit = {
+    if (Files.isDirectory(f)) {
+      Files.list(f).iterator.asScala.foreach(delete)
+    }
+    Files.delete(f)
+  }
+
+  def execTest[T,A](makeAction: EphemerisFiles => TryExport[T])(processResult: (ExportError \/ T) => A): A = {
+    val dir = Files.createTempDirectory("EphemerisFilesTest-")
+    val ef  = EphemerisFiles(dir)
+
+    try {
+      processResult(makeAction(ef).run.unsafePerformIO())
+    } finally {
+      delete(dir)
+    }
+  }
+
+  def testLeft[T](makeAction: EphemerisFiles => TryExport[T]): ExportError =
+    execTest(makeAction) {
+      case -\/(err) => err
+      case \/-(t)   => throw new RuntimeException("Expected error")
+    }
+
+  def testRight[T](makeAction: EphemerisFiles => TryExport[T]): T =
+    execTest(makeAction) {
+      case -\/(err) =>
+        val (msg, ex) = err.report
+        log.log(Level.WARNING, msg, ex.orNull)
+        err.log(log)
```

Comment:
Oops, it looks like I'm logging the test case error twice.  Thanks, I'll fix that.